dali/pipeline/data/buffer.h

```diff
@@ -307,7 +326,45 @@ class DLL_PUBLIC Buffer {
     return shares_data_;
   }
 
-  DISABLE_COPY_MOVE_ASSIGN(Buffer);
+  /**
+   * @brief Set another Buffer as the backing memory for this Buffer.
+   *
+   * Current Buffer will be marked as sharing data, and reallocation of memory will be
+   * prohibited until reset() is called.
+   */
+  inline void set_backing_allocation(const Buffer<Backend> &other) {
```

Comment:
Why not `ShareData`? That's what we use in Tensor and TensorXxx - it's not that different here. After all, this does more than just setting the backing allocation - it copies other properties, too (type, logical size, device).

Reply:
I think this names is more descriptive compared to ShareData - ShareData suggests that it can work kinda both ways, but maybe I'm reading too much into it. If you set some external allocation to this Buffer (be it some shared_ptr or other Buffer), we assume it's valid until you try to cause a reallocation.

Reply:
Reverted to ShareData for the Buffer overload.